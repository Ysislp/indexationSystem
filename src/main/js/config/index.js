export const localhostURL='/api/content';