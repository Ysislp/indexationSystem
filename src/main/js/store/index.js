import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import results from '../reducers/resultsReducer';
import index from '../reducers/indexReducer';

const initialState = {};

const reducer = combineReducers ({
  results,
  index
});

const store = createStore(reducer, initialState, applyMiddleware(thunk));

export default store;
