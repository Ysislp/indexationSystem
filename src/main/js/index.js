import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Results from './components/results';
import Form from './components/form';
import store from './store';

const Root = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Switch>
        <Route exact path="/" component={ Form } />
        <Route path="/all" component={ Results } />
      </Switch>
    </BrowserRouter>
  </Provider>
)

ReactDOM.render(<Root/>, document.getElementById('root'));