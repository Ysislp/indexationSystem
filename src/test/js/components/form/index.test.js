import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FormApp from '../../../../main/js/components/form';
import Page from '../../../../main/js/components/form/page';
import { regExpUrl } from '../../../../main/js/constants';

Enzyme.configure({ adapter: new Adapter() });

describe('Results component Test', () => {

  let props;

  beforeEach(() => {
    const props = {
      onChangeText: jest.fn(),
      handleSubmit: jest.fn(),
      validateUrl: jest.fn(),
      word: "",
      url: "",
      waiting: false,
      validUrl: true,
    }
  });

  it('Snapshot Form', () => {
    const form = renderer.create(
      <MemoryRouter>
        <FormApp />
      </MemoryRouter>
    )

    expect(form.toJSON()).toMatchSnapshot();
  });

  it('Should render text fields', () => {
    const component = shallow(
      <Page 
        {...props}
      />
    );

    expect(component.find(`[data-test='url']`).exists()).toBeTruthy();
    expect(component.find(`[data-test='word']`).exists()).toBeTruthy();
  });

  it('Should onSubmit handler', () => {
    const fakeEvent = { preventDefault: () => 'preventDefault' };
    const handleSubmit = jest.fn();
    const component = shallow(
      <Page 
        handleSubmit={handleSubmit} 
      />
    );
    const wrapper = component.find(`[data-test='form']`);

    wrapper.simulate('submit', fakeEvent);
    expect(handleSubmit).toBeCalled();
  });

  it('Should onBlur handler', () => {
    const validateUrl = jest.fn();
    const component = shallow(
      <Page 
        validateUrl={validateUrl}
      />
    );
    const wrapper = component.find(`[data-test='url']`);

    wrapper.simulate('blur');
    expect(validateUrl).toBeCalled();
  });

  it('Should onChange handler', () => {
    const fakeEvent = { target: { value: "sometext" } };
    const onChangeText = jest.fn();
    const component = shallow(
      <Page 
        onChangeText={onChangeText} 
      />
    );
    const wrapperUrl = component.find(`[data-test='url']`);
    const wrapperWord = component.find(`[data-test='word']`);

    wrapperUrl.simulate('change', fakeEvent);
    wrapperWord.simulate('change', fakeEvent);
    expect(onChangeText).toBeCalled();
  });

  it('Should onChangeText()', () => {
    const fakeEvent = { target: {id:"url", value:"http://google.com"} };
    const component = shallow( <FormApp /> );
    const instance = component.instance();

    instance.onChangeText(fakeEvent);
    expect(component.state(fakeEvent.target.id)).toBeTruthy();
  });

  it('Should validateUrl()', () => {
    const component = shallow( <FormApp /> );
    const regexUrl = new RegExp(regExpUrl);
    const url = "http://google.com";
    const notUrl = "something";
    const instance = component.instance();

    instance.validateUrl();
    expect(regexUrl.test(url)).toBe(true);
    expect(regexUrl.test(notUrl)).toBe(false);
  });

  it('Should handleSubmit()', () => {
    const fakeEvent = { preventDefault: () => 'preventDefault' };
    const data = JSON.stringify({
      word: "test", 
      url: "http://google.com"
    });
    const component = shallow( <FormApp /> );
    const instance = component.instance();

    instance.handleSubmit(fakeEvent);
    expect(component.state('waiting')).toEqual(true);
  });
  
})