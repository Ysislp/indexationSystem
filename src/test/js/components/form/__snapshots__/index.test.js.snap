// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component Test Snapshot Form 1`] = `
<div
  className="makeStyles-body-1"
>
  <div
    className="makeStyles-root-7"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-menuButton-8"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
          />
        </svg>
        <h6
          className="MuiTypography-root makeStyles-title-9 MuiTypography-h6"
        >
          Indexation System
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text button MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="makeStyles-links-10"
              href="/all"
              onClick={[Function]}
            >
              Ver Todos
            </a>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text button MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="makeStyles-links-10"
              href="/"
              onClick={[Function]}
            >
              Agregar
            </a>
          </span>
        </button>
      </div>
    </header>
  </div>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <form
      className="makeStyles-form-3"
      data-test="form"
      onSubmit={[Function]}
    >
      <div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-field-5"
          data-test="url"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
            htmlFor="url"
            id="url-label"
          >
            URL
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="url"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-field-5"
          data-test="word"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink={false}
            htmlFor="word"
            id="word-label"
          >
            Word
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="word"
              maxLength={15}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-myButton-4"
        data-test="submitButton"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
    </form>
  </div>
</div>
`;
